pkg/resourcemanager/psql/psqluser/psqluser_reconcile.go

```diff
@@ -0,0 +1,446 @@
+			// Add the format to the output map if it has been requested otherwise call for its deletion from the secret store
+			if requested {
+				switch formatName {
+				case "adonet":
+					formattedSecrets["adonet"] = []byte(fmt.Sprintf(
+						"Server=tcp:%v,1433;Initial Catalog=%v;Persist Security Info=False;User ID=%v;Password=%v;MultipleActiveResultSets=False;Encrypt=True;TrustServerCertificate=False;Connection Timeout=30;",
+						string(DBSecret["fullyQualifiedServerName"]),
+						instance.Spec.DbName,
+						user,
+						string(DBSecret["password"]),
+					))
+
+				case "adonet-urlonly":
+					formattedSecrets["adonet-urlonly"] = []byte(fmt.Sprintf(
+						"Server=tcp:%v,1433;Initial Catalog=%v;Persist Security Info=False; MultipleActiveResultSets=False;Encrypt=True;TrustServerCertificate=False;Connection Timeout",
+						string(DBSecret["fullyQualifiedServerName"]),
+						instance.Spec.DbName,
+					))
+
+				case "jdbc":
+					formattedSecrets["jdbc"] = []byte(fmt.Sprintf(
+						"jdbc:sqlserver://%v:1433;database=%v;user=%v@%v;password=%v;encrypt=true;trustServerCertificate=false;hostNameInCertificate=*."+config.Environment().SQLDatabaseDNSSuffix+";loginTimeout=30;",
+						string(DBSecret["fullyQualifiedServerName"]),
+						instance.Spec.DbName,
+						user,
+						instance.Spec.Server,
+						string(DBSecret["password"]),
+					))
+				case "jdbc-urlonly":
+					formattedSecrets["jdbc-urlonly"] = []byte(fmt.Sprintf(
+						"jdbc:sqlserver://%v:1433;database=%v;encrypt=true;trustServerCertificate=false;hostNameInCertificate=*."+config.Environment().SQLDatabaseDNSSuffix+";loginTimeout=30;",
+						string(DBSecret["fullyQualifiedServerName"]),
+						instance.Spec.DbName,
+					))
+
+				case "odbc":
+					formattedSecrets["odbc"] = []byte(fmt.Sprintf(
+						"Server=tcp:%v,1433;Initial Catalog=%v;Persist Security Info=False;User ID=%v;Password=%v;MultipleActiveResultSets=False;Encrypt=True;TrustServerCertificate=False;Connection Timeout=30;",
+						string(DBSecret["fullyQualifiedServerName"]),
+						instance.Spec.DbName,
+						user,
+						string(DBSecret["password"]),
+					))
+				case "odbc-urlonly":
+					formattedSecrets["odbc-urlonly"] = []byte(fmt.Sprintf(
+						"Driver={ODBC Driver 13 for SQL Server};Server=tcp:%v,1433;Database=%v; Encrypt=yes;TrustServerCertificate=no;Connection Timeout=30;",
+						string(DBSecret["fullyQualifiedServerName"]),
+						instance.Spec.DbName,
+					))
+				case "server":
+					formattedSecrets["server"] = DBSecret["fullyQualifiedServerName"]
+
+				case "database":
+					formattedSecrets["database"] = []byte(instance.Spec.DbName)
+
+				case "username":
+					formattedSecrets["username"] = []byte(user)
+
+				case "password":
+					formattedSecrets["password"] = DBSecret["password"]
+				}
+			} else {
+				err = sqlUserSecretClient.Delete(
+					ctx,
+					types.NamespacedName{Namespace: key.Namespace, Name: instance.Name + "-" + formatName},
+				)
+			}
+		}
+
```

Comment:
I'd like to drop all this secret formatting stuff. There is a PR to refactor it in AzureSQL and I don't want to introduce this feature until its fixed and/or required by a user.